HeartbeatLogging/Sources/Weak.swift

```diff
@@ -0,0 +1,22 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+// TODO: Document.
+
+/// <#Description#>
+struct Weak<Object: AnyObject> {
```

Comment:
The name looks too generic to me. I would prefer more verbose like `WeakContainer` or something along these lines.

Reply:
`WeakContainer` is good. Would you prefer `WeakReference` or `WeakContainer`?

Reply:
Both LGTM, but I like WeakContainer slightly more.

Reply:
SGTM, thanks

Reply:
Reworked in e9d0361